[bug]: cdrel_custom :SQLite version too old: sqlite3_prepare_v3 / SQLITE_PREPARE_PERSISTENT undeclared

### Severity

Blocker

### Versions

20.19.0

### Components/Modules

cdrel_custom

### Operating Environment

centos 7

### Frequency of Occurrence

Constant

### Issue Description

Build fails in cdrel_custom/config.c:876 because the system SQLite version predates 3.20.0 (released 2017-08-01), which is when sqlite3_prepare_v3 and the SQLITE_PREPARE_PERSISTENT flag were introduced. The compiler reports an implicit function declaration warning for sqlite3_prepare_v3 and an undeclared identifier error for SQLITE_PREPARE_PERSISTENT, causing the build to abort.


### Relevant log output

```shell
cdrel_custom/config.c: In function ‘open_database’:
cdrel_custom/config.c:876:2: warning: implicit declaration of function ‘sqlite3_prepare_v3’ [-Wimplicit-function-declaration]
  res = sqlite3_prepare_v3(config->db, sql, -1, SQLITE_PREPARE_PERSISTENT, &config->insert, NULL);
  ^
cdrel_custom/config.c:876:48: error: ‘SQLITE_PREPARE_PERSISTENT’ undeclared (first use in this function)
  res = sqlite3_prepare_v3(config->db, sql, -1, SQLITE_PREPARE_PERSISTENT, &config->insert, NULL);
                                                ^
cdrel_custom/config.c:876:48: note: each undeclared identifier is reported only once for each function it appears in
make[1]: *** [cdrel_custom/config.o] Error 1
make: *** [res] Error 2
```

### Asterisk Issue Guidelines

- [x] Yes, I have read the Asterisk Issue Guidelines

--- a/res/cdrel_custom/config.c
+++ b/res/cdrel_custom/config.c
@@ -873,7 +873,11 @@ static int open_database(struct cdrel_config *config)
 		return -1;
 	}
 
+#if SQLITE_VERSION_NUMBER >= 3020000
 	res = sqlite3_prepare_v3(config->db, sql, -1, SQLITE_PREPARE_PERSISTENT, &config->insert, NULL);
+#else
+	res = sqlite3_prepare_v2(config->db, sql, -1, &config->insert, NULL);
+#endif
 	if (res != SQLITE_OK) {
 		ast_log(LOG_ERROR, "%s->%s: Unable to prepare INSERT statement '%s': %s\n",
 			cdrel_basename(config->config_filename), cdrel_basename(config->output_filename),

